### PR TITLE
NIFI-6734: S3EncryptionService fixes and improvements

### PR DESCRIPTION
**Pull request by turcsanyip**
This PR is intentionally not a single commit to make the review easier.

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [x] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/s3/AbstractS3Processor.java [hunk 1]
```diff
@@ -130,8 +130,9 @@
     public static final PropertyDescriptor ENCRYPTION_SERVICE = new PropertyDescriptor.Builder()
             .name("encryption-service")
             .displayName("Encryption Service")
-            .description("Specifies the Encryption Service Controller used configure requests.  "
-                    + "For backward compatibility, this value is ignored when 'Server Side Encryption' is set.")
+            .description("Specifies the Encryption Service Controller used configure requests. " +
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/s3/AbstractS3Processor.java]: Typo: "used _to_ configure requests. "

**Comment by natural** [comment]
This looks great, @turcsanyip, thanks for the help.  I don't see anything that needs to change.

**Comment by turcsanyip** [comment]
@natural Thanks for the review!

@committers: Would it be possible to keep the separate commits when it gets merged? (to preserve the history and keep it more understandable)

**Comment by alopresto** [comment]
Peter, I plan to squash the commits for readability on the master branch, but I suggest keeping your branch in existence which will allow any interested parties to revisit if they need to explore the steps taken. This is what I often do for my own feature work which includes many component steps.

**Comment by turcsanyip** [comment]
OK, that's fine with me. Thanks

File: nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/encryption/StandardS3EncryptionService.java [hunk 2]
```diff
@@ -52,74 +53,80 @@
...
             .name("key-id-or-key-material")
             .displayName("Key ID or Key Material")
-            .description("For Server-side CEK and Client-side CMK, this is base64-encoded Key Material.  For all others (except 'None'), it is the KMS Key ID.")
+            .description("For None and Server-side S3: not used. For Server-side KMS and Client-side KMS: the KMS Key ID must be configured. " +
+                    "For Server-side Customer Key and Client-side Customer Key: the Key Material must be specified in Base64 encoded form. " +
+                    "In case of Server-side Customer Key, the key must be an AES-256 key. In case of Client-side Customer Key, it can be an AES-256, AES-192 or AES-128 key.")
             .required(false)
             .sensitive(true)
-            .addValidator(new StandardValidators.StringLengthValidator(0, 4096))
+            .addValidator((subject, input, context) -> new ValidationResult.Builder().valid(true).build()) // will be validated in customValidate()
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/s3/encryption/StandardS3EncryptionService.java]: If `customValidate()` validates this value, it does not need a static validator here.

**Pull request closed without merging**